articles/how-to-prevent-cross-site-scripting-in-node.js/index.md

```diff
@@ -0,0 +1,133 @@
+
+All this time, the user is unaware that an attack is happening. That is how XSS attacks work.
+
+Cross-site scripting vulnerabilities are:
+- Easy to spot
+- Easy to use
+- Have a high impact on your business security
+
+### Types of XSS attacks
+Below are the different types of XSS attacks.
+
+#### Reflected cross-site scripting
+[Reflected XSS](https://portswigger.net/web-security/cross-site-scripting/reflected) (Non-persistent XSS) is an attack that occurs when a malicious script is redirected to an application via the HTTP. In non-persistent XSS, the malicious code is executed instantly in the victim's browser. 
+
+If the server fails to encode the user inputs properly, the attacker might search for such a string `<script>alert (‘XSS!')</script>`. This JavaScript code reflects on the browser as part of the code. The code is then executed to display an alert that confirms XSS vulnerability. Below is an example of a reflected XSS attack.
+
+The HTTP link before the malicious code is injected
+`https://example-webapp.com/report =Safe`
+
+How the attack is performed,
+`https://example-webapp.com/report =<script>\/*Not+Safe…?+*/</script>`.
+
+When the user accesses the URL sent by the attacker, the malicious script executes immediately in the browser.
+
+#### Stored cross-site scripting
+[Stored XSS](https://portswigger.net/web-security/cross-site-scripting/stored) (Persistent XSS) occurs when a malicious script is saved in the website's database. The malicious script executes when the user accesses the web app. At any moment the victim opens the browser, the malicious script runs.
+
+#### DOM-based cross-site scripting
+It is an attack in the DOM (Document Object Model). It arises when a malicious script executes due to DOM modification in a user's browser.
+Most developers do not understand how [DOM-based XSS](https://portswigger.net/web-security/cross-site-scripting/dom-based) works.
+
+### How to prevent XSS
+XSS is the leading attack for web applications. Now that we have an understanding of XSS and how it damages applications. The following are techniques used to [prevent XSS vulnerability](https://cheatsheetseries.owasp.org/cheatsheets/Cross_Site_Scripting_Prevention_Cheat_Sheet.html) in Node.js
+
+#### Output Encoding
+Output encoding translates untrusted input into a safe model. It takes place by running a JavaScript entity encoding in a Node.js platform server prior to sending untrusted data into the browser. 
+
+#### Avoid Inserting Untrusted Data
+Do not pass untrusted data into executable JavaScript functions such as `eval`. In the Node.js `eval` function, enable code injections. The `eval` function takes a string argument and interprets it as a JavaScript code allowing the attacker to execute it.
```

Comment:
This part is not clear. What data and what is the role of the function `eval`? What are code injections? Explain it in the proper order.